pennylane/devices/qubit/simulate.py

```diff
@@ -456,20 +446,19 @@ def parse_native_mid_circuit_measurements(
     Returns:
         tuple(TensorLike): The results of the simulation
     """
-    idx_one_shot = find_not_measurement_values(circuit)
-    normalized_meas = [None] * len(circuit.measurements)
-    for i, m in zip(idx_one_shot, all_shot_meas):
-        normalized_meas[i] = gather_non_mcm(circuit.measurements[i], m, mcm_shot_meas)
-
     idx_sample = find_measurement_values(circuit)
-    counter = Counter()
-    if any(isinstance(m, (CountsMP, ProbabilityMP)) for m in circuit.measurements):
-        for d in mcm_shot_meas:
-            counter.update(d)
-    for i in idx_sample:
-        normalized_meas[i] = gather_mcm(
-            circuit.measurements[i], circuit.measurements[i].mv, mcm_shot_meas, counter
-        )
+    normalized_meas = [None] * len(circuit.measurements)
+    for i, m in enumerate(circuit.measurements):
+        if not isinstance(m, (CountsMP, ExpectationMP, ProbabilityMP, SampleMP, VarianceMP)):
+            raise ValueError(
+                f"Native mid-circuit measurement mode does not support {type(m).__name__} measurements."
+            )
+        if i in idx_sample:
```

Comment:
```suggestion
        if m.mv:
```

I feel like I'm missing something. This should be the same, right?

Reply:
Yes, better.